### DISCUSSION TITLE
How to return a response with status codes using plugins

### DISCUSSION THREAD

**Discussion by brijesh-onmo** [dotansimha/envelop#2301] [Q&A]
I am rerouting the request from OnExecute to a different lambda and getting the response from the lambda and setting the response using setResultAndStopExecution. However I don't see a way to set the response status code as well.
The lambda might return a 500 response.

```
async onExecute({ args, setResultAndStopExecution }): Promise<void> {
    const operationType = args.document.definitions[0].operation;

    if (<some conditions> && operationType === "mutation") {
      try {
        const response = await fetch(primaryRegionEndpoint, {
          method: args.contextValue.request.method,
          headers: args.contextValue.request.headers,
          body: JSON.stringify(args.contextValue.params),
        });
        const data = await response.json();
        setResultAndStopExecution(data);
      } catch (error) {
        const response = { error: `Failed to forward request, Error: ${String(error)}` };
        setResultAndStopExecution(response);
      }
    }
  }
```

**Comment by EmrysMyrddin** [comment]
Hi, sorry for the delay...

I suppose you are using Yoga ? If it's the case, you can customize HTTP code and headers when an error is part of the result by using error extensions: https://the-guild.dev/graphql/yoga-server/docs/features/error-masking#modifying-http-status-codes-and-headers
